src/super_gradients/training/samples/detection_sample.py

```diff
@@ -0,0 +1,78 @@
+import dataclasses
+import numpy as np
+import torch
+
+from typing import Optional, List, Union
+
+from super_gradients.training.datasets.data_formats.bbox_formats.xywh import xywh_to_xyxy, xyxy_to_xywh
+
+__all__ = ["DetectionSample"]
+
+
+@dataclasses.dataclass
+class DetectionSample:
+    """
+    A data class describing a single object detection sample that comes from a dataset.
+    It contains both input image and target information to train an object detection model.
+
+    :param image:              Associated image with a sample. Can be in [H,W,C] or [C,H,W] format
+    :param bboxes_xywh:        Numpy array of [N,4] shape with bounding box of each instance (XYWH)
+    :param labels:             Numpy array of [N] shape with class label for each instance
+    :param is_crowd:           (Optional) Numpy array of [N] shape with is_crowd flag for each instance
+    :param additional_samples: (Optional) List of additional samples for the same image.
+    """
+
+    __slots__ = ["image", "bboxes_xywh", "labels", "is_crowd", "additional_samples"]
+
+    image: Union[np.ndarray, torch.Tensor]
+    bboxes_xywh: np.ndarray
+    labels: np.ndarray
+    is_crowd: Optional[np.ndarray]
+    additional_samples: Optional[List["DetectionSample"]]
+
+    def sanitize_sample(self) -> "DetectionSample":
+        """
+        Apply sanity checks on the detection sample, which includes:
+        - Clamp bbox coordinates to ensure they are within image boundaries
+        This function does not remove instances, but may make them subject for removal instead.
+        :return: self
+        """
+        image_height, image_width, _ = self.image.shape
+
+        # Clamp bboxes to image boundaries
+        clamped_boxes = xywh_to_xyxy(self.bboxes_xywh, image_shape=(image_height, image_width))
+        clamped_boxes[..., [0, 2]] = np.clip(clamped_boxes[..., [0, 2]], 0, image_width - 1)
+        clamped_boxes[..., [1, 3]] = np.clip(clamped_boxes[..., [1, 3]], 0, image_height - 1)
+        clamped_boxes = xyxy_to_xywh(clamped_boxes, image_shape=(image_height, image_width))
+
+        self.bboxes_xywh = clamped_boxes
+        return self
+
+    def filter_by_mask(self, mask: np.ndarray) -> "DetectionSample":
+        """
+        Remove pose instances with respect to given mask.
```

Comment:
pose instances -> object instances ?

Reply:
Fixed in e64d3e98e1fed5bcce5074f88845d615737f23d0 thanks!